### DISCUSSION TITLE
What to use for Microsoft Login, Azure AD or Azure AD B2C?

### DISCUSSION THREAD

**Discussion by jcocodeit** [nextauthjs/next-auth#8748] [Help]
### Summary

Hi, the title pretty much says it all.

I am trying to implement `Login with Microsoft` but I am not sure which one to use between Azure AD or Azure AD B2C.

My questions are as follows:
1. Which one to use for authenticating with Microsoft?
2. What's the difference between **Azure AD** and **Azure AD B2C**?

Thanks in advance!

**Comment by nwo31** [comment]
Were you able to solve your problem @iamcarlo042? My understanding is that if you want to authenticate for internal use, Azure AD is what you should use. If you want to authenticate external users (consumers) outside of your company, then I think you should use Azure AD B2C.
